src/hooks/useCreditForm.jsx

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+import { validateInput } from '@utils/validateInput';
+
+/**
+ *
+ * @hook useCreditForm
+ * @description 크레딧Input관련 로직을 관리하는 커스텀훅
+ *
+ * @param {number} credit 사용자의 현재보유크레딧
+ * @param {boolean} isDonate CreditForm에서 내려받은 isDonate Boolean값(validate검증용)
+ *
+ * @returns
+ */
+
+export const useCreditForm = (credit, isDonate = false) => {
+  const [input, setInput] = useState('');
+  const [error, setError] = useState(false);
+  const [errMsg, setErrMsg] = useState('');
```

Comment:
state가 많아질 때에는 하나의 객체 값으로 관리해주시는 것도 방법입니다. 특히 form을 구현할 때는 다음과 같이 많이 구현합니다!
```suggestion
    const [formState, setFormState] = useState({
        input: '',
        error: {
            hasError: false,
            message: ''
        }
```